lib/Pool.js

```diff
@@ -255,6 +259,13 @@ Pool.prototype.escapeId = function escapeId(value) {
   return mysql.escapeId(value, false);
 };
 
+Pool.prototype.format = function(sql, values) {
+  if (typeof this.config.connectionConfig.queryFormat == "function") {
+    return this.config.connectionConfig.queryFormat.call(this, sql, values, this.config.connectionConfig.timezone);
```

Comment:
Otherwise there is no need to add a new commit, really. I've already adjusted your commit to make it an instance method (still under your name). The problem right now is backwards compatibility with this function context.